### ISSUE TITLE
Make spacing around punctuation configurable in pretty printer

### ISSUE THREAD

**Issue by mboes** [chrisdone/hindent#112]
`HIndent.Pretty` currently hardcodes a few choices that ought to be left up to individual styles. In particular, `commas` prints something like `a,b,c,d`, where say the JohanTibell style would advocate `a, b, c, d` (there's an example of this in his style guide, and certainly this is what the authors @tibbe cites as influences do). The current JohanTibell style works around this in expressions, but it's hard to work around this in general. For example, that style currently prints

```Haskell
f :: (Eq a,Show a) => a -> (a,a,a)
f x = (x, x, x)
```

which is inconsistent.

**Comment by mboes** [comment]
Ah! Figured out why I couldn't manage to override the context pretty printing. I guess this ticket is now more a problem with the current johantibell style implementation, than with the pretty printer.

**Comment by chrisdone** [comment]
Good catch. Indeed, type tuples and context lists should be spaced. Did you manage to configure that?
